lib/compress.js

```diff
@@ -2611,7 +2611,8 @@ merge(Compressor.prototype, {
                 if (compressor.option("unused")
                     && def.references.length == 1
                     && compressor.find_parent(AST_Scope) === def.scope) {
-                    if (!compressor.option("keep_fnames")) {
+                    if (!compressor.option("keep_fnames")
+                        && exp.name && exp.name.definition() === def) {
                         exp.name = null;
```

Comment:
Ah, it's because the original copy of the function is still around and waiting for elimination by `drop_unused()`, so it can't tell the function is no longer referenced afterwards.

I'll think about this and give it a proper fix this afternoon.